kong2kic/route.go

```diff
@@ -256,3 +259,288 @@ func addPluginsToRoute(
 	}
 	return nil
 }
+
+// HeadersByNameTypeAndValue is a type to sort headers by name, type and value.
+// This is needed to ensure that the order of headers is consistent across runs.
+type HeadersByNameTypeAndValue []k8sgwapiv1.HTTPHeaderMatch
+
+func (a HeadersByNameTypeAndValue) Len() int      { return len(a) }
+func (a HeadersByNameTypeAndValue) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a HeadersByNameTypeAndValue) Less(i, j int) bool {
+	if a[i].Name < a[j].Name {
+		return true
+	}
+	if a[i].Name > a[j].Name {
+		return false
+	}
+
+	if a[i].Type != nil && a[j].Type == nil {
+		return true
+	}
+	if a[i].Type == nil && a[j].Type != nil {
+		return false
+	}
+
+	if *a[i].Type < *a[j].Type {
+		return true
+	}
+	if *a[i].Type > *a[j].Type {
+		return false
+	}
+	return a[i].Value < a[j].Value
+}
+
+// Convert route to HTTPRoute (Gateway API)
+func populateKICIngressesWithGatewayAPI(content *file.Content, kicContent *KICContent) error {
+	for _, service := range content.Services {
+		for _, route := range service.Routes {
+			httpRoute, err := createHTTPRoute(service, route)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+
+			err = addPluginsToGatewayAPIRoute(service, route, httpRoute, kicContent)
+			if err != nil {
+				return err
+			}
+			kicContent.HTTPRoutes = append(kicContent.HTTPRoutes, httpRoute)
+		}
+	}
+	return nil
+}
+
+func createHTTPRoute(service file.FService, route *file.FRoute) (k8sgwapiv1.HTTPRoute, error) {
+	var httpRoute k8sgwapiv1.HTTPRoute
+	httpRoute.Kind = "HTTPRoute"
+	if targetKICVersionAPI == KICV3GATEWAY {
+		httpRoute.APIVersion = GatewayAPIVersionV1
+	} else {
+		httpRoute.APIVersion = GatewayAPIVersionV1Beta1
+	}
+	if service.Name != nil && route.Name != nil {
+		httpRoute.ObjectMeta.Name = calculateSlug(*service.Name + "-" + *route.Name)
```

Comment:
Are we sure that `httpRoute.ObjectMeta` is always non-nil?
I am unaware about the underlying struct but if it ever comes nil, our code can panic here.

Reply:
ObjectMeta is a struct so it is always initialized.